### PR TITLE
refactor(form): convert LESS styles to SCSS

### PR DESCRIPTION
**Pull request by 100stacks**
* Refactor `form` LESS styles to SCSS

### LGTM's
* [x] Dev LGTM
* [x] Dev LGTM

### CODE REVIEW

File: src/scss/components/form/_index.scss [hunk 1]
```diff
@@ -1 +1,157 @@
...
+
+  If we apply `content: '*'` to `::before`, a screen reader would
+  announce "star" as if it were part of the coded label text.
+
+  Obviously, this is underiable.
+
+  By inlining an SVG of the asterisk into `background-image`, we can
+  provide sighted users an indicator while allowing screen readers to
+  correctly announce text content (as they already do).
+*/
```
Comment (lalithkarikelli) [inline src/scss/components/form/_index.scss]: For bulk comment please follow "///" to the complete section

File: src/less/components.less [hunk 2]
```diff
@@ -23,4 +23,3 @@
     - Complex CSS Selectors
     - NO !important
 \* ========== COMPONENTS ========== */
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/less/components.less]: Comment removed.

File: src/scss/components/form/_index.scss [hunk 3]
```diff
@@ -1 +1,157 @@
...
+.beta-hxField {
+  // placeholder for potential styles
+}
+
+.beta-hxFieldName {
+  @include hxFieldName;
+}
+//end:COMPONENT
+
+/* ---------- FORM HELPERS ---------- */
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/scss/components/form/_index.scss]: Comment format updated.

File: src/less/components.less (hunk 2 above)
[the remark this answers is not in the document]
Comment (100stacks) [reply src/less/components.less]: Comment removed.

[the remark this answers is not in the document]
Comment (100stacks) [reply src/less/components.less]: Comment removed.

File: src/scss/components/form/_index.scss (hunk 1 above)
Comment (100stacks) [reply src/scss/components/form/_index.scss]: Ok comment updated.

File: src/scss/components/form/_index.scss [hunk 4]
```diff
@@ -1 +1,156 @@
+@import "config";
+
+// for Firefox
 @import "reset"; // ./_reset.scss
+
+
```
Comment (BY00565233) [inline src/scss/components/form/_index.scss]: Please remove the extra white space line number :6

File: src/scss/components/form/_index.scss [hunk 5]
```diff
@@ -1 +1,157 @@
...
+.hxOptional {
+  &::after {
+    color: $gray-700;
+    content: " (optional)";
+  }
+}
+
+// DEPRECATED
+@import "./deprecated"; // remove in v1.0.0
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/scss/components/form/_index.scss]: Yes, we normally include all import statements at the top of the file.

For deprecated CSS, the pattern for this file is put the deprecated `@import` at the bottom of the file.  We can revisit this pattern later.

File: src/scss/components/form/_index.scss (hunk 4 above)
Comment (100stacks) [reply src/scss/components/form/_index.scss]: Extra space removed.

File: src/less/components/form/FieldName.less [hunk 6]
```diff
@@ -1,12 +0,0 @@
-// ===================================== //
```
Comment (100stacks) [inline src/less/components/form/FieldName.less]: This file - `src/less/components/form/FieldName.less` - refactored in PR #589.

File: src/less/components/form/TextControl.less [hunk 7]
```diff
@@ -1,63 +0,0 @@
-// ===================================== //
```
Comment (100stacks) [inline src/less/components/form/TextControl.less]: This file - `src/less/components/form/TextControl.less` - refactored in PR #589.

**Pull request merged**